features/steps/steps.py

```diff
@@ -1,33 +1,294 @@
+
+def get_temporary_output_filename():
+    return 'temp_doi_label.xml'
+
+def save_doi_to_temporary_file(doi_label):
+    # Save doi_label to disk so it can be compared to historical in next step.
+    temporary_file_name = get_temporary_output_filename()
+    temporary_file_ptr = open(temporary_file_name,"w+") 
+    temporary_file_ptr.write(doi_label + "\n")
+    temporary_file_ptr.close()
+    return temporary_file_name
+
+def draft_action_run(node_value,input_value):
+    # Helper function to 'draft' a given input_value and write the DOI to a temporary file.
+    # This file will be available for other validation functions.
+    db_name = 'doi_temp.db'
+    _action = DOICoreActionDraft(db_name=db_name)
+    logger.info(f"input_value {input_value}")
+
+    o_doi_label = _action.run(input=input_value,
+                          node=node_value,
+                          submitter='my_user@my_node.gov',force=True)
+    # Save o_doi_label to disk so it can be compared to historical in next step
+    logger.info(f"success input_value {input_value}")
+    return save_doi_to_temporary_file(o_doi_label)
+
+def reserve_action_run(node_value,input_value):
+    # Helper function to 'reserve' a given input_value.
+    global g_action_expectant_bad
+    logger.info(f"when node_value,input_value {node_value,input_value}")
+    o_doi_label = None
+
+    db_name = 'doi_temp.db'
+    _action = DOICoreActionReserve(db_name=db_name)
+
+    o_doi_label = _action.run(
+                      input=input_value,
+                      node=node_value, submitter='my_user@my_node.gov',
+                      dry_run=True,force=True)
+
+    return save_doi_to_temporary_file(o_doi_label)
+    #return o_doi_label
+
+def draft_output_compare(output_file, ref_output_value):
+    # Function compare two XML files created from 'draft' or 'reserve' actions.
+    # Assumption(s): 
+    #   1.  The name of the new XML file is defined in get_temporary_output_filename().
+    #   2.  The name of the historical name is ref_output_value
+    logger.info(f"output_file,ref_output_value {output_file},{ref_output_value}")
+
+    o_fields_differ_list, o_values_differ_list, o_record_index_differ_list = DOIDiffer.doi_xml_differ(ref_output_value,
+                                                                                                      output_file)
+
+
+    logger.info(f'different fields are {o_fields_differ_list}')
+
+    assert len(o_fields_differ_list) is 0
+
+    return 1
 
-@given('a valid PDS4 {input_type} at url {input_value}')
-def given_valid_input(context, input_type, input_value):
-    logger.info(f"given {input_type} {input_value}")
+def reserve_output_compare(output_file, ref_output_value):
+    logger.info(f"output_file,ref_output_value {output_file},{ref_output_value}")
+    # Use the same function draft_output_compare() to compare a DOI (XML output) from the 'reserve' action.
+    draft_output_compare(output_file,ref_output_value)
```

Comment:
The function started out comparing the output from draft and found it can be reused for reserve.

It has been renamed to file_output_compare now to be more generic.

Reply:
Ok fine, I removed the reserve_output_compare function since it is the same as file_output_compare